opm/autodiff/BlackoilPropsAdFromDeck.cpp

```diff
@@ -166,116 +168,35 @@ BlackoilPropsAdFromDeck::BlackoilPropsAdFromDeck(const BlackoilPropsAdFromDeck&
 
         phase_usage_ = phaseUsageFromDeck(deck);
 
+        gasPvt_ = std::make_shared<GasPvt>();
+        oilPvt_ = std::make_shared<OilPvt>();
+        waterPvt_ = std::make_shared<WaterPvt>();
 
+        gasPvt_->initFromDeck(deck, eclState);
```

Comment:
In my opinion, if you does not specify the keyword `GAS` in the deck, it means no gas phase will be involved in the simulation.  I am not 100% sure about this opinion though.